### ISSUE TITLE
How can I run my Python scraping script without needing to install Python? (Simple solution for a non-programming user)

### ISSUE THREAD

**Issue by ALeterouin** [ALeterouin/Booking_scraper#4]
Hello

I am working on a Python script that scrapes hotel data from Booking.com. The script uses libraries such as Selenium, Pandas, and Tkinter to gather hotel information and save it into an Excel file. I’ve uploaded everything to a GitHub repository: https://github.com/ALeterouin/Booking_scraper, but I’m wondering how to make it so that my friend, who doesn’t know how to code and doesn’t have Python installed on their computer, can easily run the script without any difficulties.

Here’s my situation:

My friend doesn’t know Python and doesn’t want to bother installing a full Python environment.

They just want to be able to download my code from GitHub or have me send them the file directly, and then run it on their computer.

What I’ve tried:

I thought about providing an .exe file or using tools like PyInstaller or cx_Freeze, but I’ve encountered some issues related to different versions of the operating systems. I might have gone about it the wrong way.

Questions:

How can I create an executable version of my Python script that can be run directly, without requiring Python?

Are there any simple and lightweight alternatives for my friend to run this script without having to set up a Python environment?

Thank you in advance for your advice and solutions!

@AhmedMansour024

**Comment by AhmedMansour024** [comment]
i saw you post on Reddit but i really dont know how to do it.

**Comment by AhmedMansour024** [comment]
he can run the code without a code editor but he have to install python.

**Comment by ALeterouin** [comment]
Yeah i know but for someone who never dl python it's very hard. And i can't install for him.

**Comment by AhmedMansour024** [comment]
you can use .md file to guide him or use Makefile to Automate the installation but it needs to be installed to.

**Comment by ALeterouin** [comment]
Ok, thank you. I will try to improve my code.
Do you needs help for anythings? im trying to be better in python

**Comment by AhmedMansour024** [comment]
iam getting an error when running your code!
![Capture](https://github.com/user-attachments/assets/9c54925a-0872-4ee4-856f-c9a852f5c516)

**Comment by ALeterouin** [comment]
It's probably because im using a mac. Idk how to help you
